Managers/WindowMan.cpp

```diff
@@ -0,0 +1,602 @@
+
+				if (result != 0) {
+					break;
+				}
+			}
+		} else {
+			result = SDL_RenderSetVSync(m_PrimaryRenderer.get(), sdlEnableVSync);
+		}
+#endif
+
+		if (result != 0) {
+			ShowMessageBox("Unable to change VSync mode at runtime! The change will be applied after restarting!");
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void WindowMan::UpdateInfoOfDisplayPrimaryWindowIsAt() {
+		m_DisplayIndexPrimaryWindowIsAt = SDL_GetWindowDisplayIndex(m_PrimaryWindow.get());
+
+		SDL_Rect currentDisplayBounds;
+		SDL_GetDisplayBounds(m_DisplayIndexPrimaryWindowIsAt, &currentDisplayBounds);
+
+		m_DisplayWidthPrimaryWindowIsAt = currentDisplayBounds.w;
+		m_DisplayHeightPrimaryWindowIsAt = currentDisplayBounds.h;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void WindowMan::MapDisplays(bool updateInfoOfDisplayPrimaryWindowIsAt) {
+		auto setSingleDisplayMode = [this](const std::string &errorMsg = "") {
+			m_MaxResX = m_DisplayWidthPrimaryWindowIsAt;
+			m_MaxResY = m_DisplayHeightPrimaryWindowIsAt;
+			m_NumDisplays = 1;
+			m_DisplayArrangementLeftMostOffset = -1;
+			m_DisplayArrangementTopMostOffset = -1;
+			m_ValidDisplayIndicesAndBoundsForMultiDisplayFullscreen.clear();
+			m_CanMultiDisplayFullscreen = false;
+			if (!errorMsg.empty()) {
+				ShowMessageBox("Failed to map displays for multi-display fullscreen because:\n\n" + errorMsg + "!\n\nFullscreen will be limited to the display the window is positioned at!");
+			}
+		};
+
+		if (updateInfoOfDisplayPrimaryWindowIsAt) {
+			UpdateInfoOfDisplayPrimaryWindowIsAt();
+		}
+
+		m_NumDisplays = SDL_GetNumVideoDisplays();
+
+		if (m_IgnoreMultiDisplays || m_NumDisplays == 1) {
+			setSingleDisplayMode();
+			return;
+		}
+
+		m_ValidDisplayIndicesAndBoundsForMultiDisplayFullscreen.clear();
+
+		int leftMostOffset = 0;
+		int topMostOffset = std::numeric_limits<int>::max();
+		int maxHeight = std::numeric_limits<int>::min();
+		//int maxUsableHeight = std::numeric_limits<int>::max();
+		//int minHeightDisplayIndex = -1;
+		int totalWidth = 0;
+
+		for (int displayIndex = 0; displayIndex < m_NumDisplays; ++displayIndex) {
+			SDL_Rect displayBounds;
+			if (SDL_GetDisplayBounds(displayIndex, &displayBounds) == 0) {
+				m_ValidDisplayIndicesAndBoundsForMultiDisplayFullscreen.emplace_back(displayIndex, displayBounds);
+
+				leftMostOffset = std::min(leftMostOffset, displayBounds.x);
+				topMostOffset = std::min(topMostOffset, displayBounds.y);
+				maxHeight = std::max(maxHeight, displayBounds.h);
+
+				/*
+				// TODO: Figure out not-quite-letterboxing-but-letterboxing in multi-display fullscreen.
+				// Deal with this by creating the backbuffer to fit the minimum height display and then offset the taller display textures on the renderers to align everything.
```

Comment:
ded code ded code ded code

In all seriousness, the bit of code snippet here doesn't seem very useful. Seems like you could just keep the todo get rid of this cruft. All else aside, it's in git history now if you ever really want it.